### ISSUE TITLE
Rule proposal: padding-lines in script tag

### ISSUE THREAD

**Issue by xyannik** [vuejs/eslint-plugin-vue#1452]
**Please describe what the rule should do:**
Enforce empty lines between blocks inside the <script> tag (blank lines between `name`, `props`, `setup` in Composition API, or between `name`, `props`, `computed`, `data`, `methods` in Options API)

**What category should the rule belong to?**

- [x]  Enforces code style (layout)
- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```js
// BAD
export default {
  name: 'ComponentName',
  props: { ... },
  setup () { ... }
}
```

```js
// GOOD
export default {
  name: 'ComponentName',

  props: { ... },

  setup () { ... }
}
```

**Additional context**
This is similar to the existing rule `vue/padding-line-between-blocks` but applies to the blocks in `<script>`
Similar to https://eslint.org/docs/rules/lines-between-class-members

**Comment by ota-meshi** [comment]
Does the `vue/new-line-between-multi-line-property` rule solve your style?
https://eslint.vuejs.org/rules/new-line-between-multi-line-property.html

**Comment by xyannik** [comment]
> Does the `vue/new-line-between-multi-line-property` rule solve your style?
> https://eslint.vuejs.org/rules/new-line-between-multi-line-property.html

Thanks! Somehow I missed that rule. It's pretty much what I wanted. Only thing that it misses is a newline after `name: 'ComponentName` since it's not multiline. But for all other purposes it works as intended, thanks :)

**Comment by tpuida** [comment]
can we add configs to make mandatory padding lines only for vue "options"?
```
// GOOD
export default {
  name: 'ComponentName',

  props: {
    value: {
      type: String,
      required: true
    },
    focused: {
      type: Boolean,
      default: false,
      required: true
    },
    label: String,
    icon: String
  },

  setup () { ... }
}```
